### DISCUSSION TITLE
Installing Pycuda On Windows 10

### DISCUSSION THREAD

**Discussion by omerXfaruq** [inducer/pycuda#332] [Troubleshooting]
Hello, I installed pycuda on my machine successfully, after installing visual studio c++ 14.0
However even if I can call nvcc from shell pycuda can not locate Cuda or nvcc.

Is there a up-to-date installation procedure in Windows 10 for pycuda?
I searched the internet and also looked into wiki page, however I couldn't find working installation for Windows 10.

Wiki page:
https://wiki.tiker.net/PyCuda/Installation/Windows 

Error:
import pycuda.driver as drv
D:\Programming Apps\Python\Python38\lib\site-packages\pycuda\driver.py:43: UserWarning: Unable to discover CUDA installation directory while attempting to add it to Python's DLL path. Either set the 'CUDA_PATH' environment variable or ensure that 'nvcc.exe' is on the path.
  warn("Unable to discover CUDA installation directory "

**Comment by omerXfaruq** [comment]
This problem is solved by adding CUDA_PATH as an independent path variable, instead of adding it to PATH

**Comment by inducer** [comment]
I'm guessing this applies *after* the package is installed. Further, the message you give is a warning, not an error. Could you be clearer on what actually goes wrong when you don't have `$CUDA_PATH` set?

**Comment by omerXfaruq** [comment]
Sorry for the inconvinience, I was trying to run a test file, there were a lot of cuda.h errors occuring.
I can not replay the same scenario, because I am getting an strange error now, path repeates \ twice.

 File "D:\Programming Apps\Python\Python38\lib\os.py", line 1109, in add_dll_directory
    cookie = nt._add_dll_directory(path)
FileNotFoundError: [WinError 2] Sistem belirtilen dosyayı bulamıyor: 'D:\\Programming Apps\\CUDA\\CUDA_v10.1\\bin\\bin'

There are very few resources on installing pycuda on Windows 10, I think it would be best for you to create a windows 10 installation tutorial if you could.

**Comment by jyhmiinlin** [comment]
Hi, I am user of pycuda. 

I wrote a tutorial for windows 10 installation. 

http://jyhmiinlin.github.io/pynufft/installation/Windows.html

Hope this helps.

**Comment by Randy37773** [comment]
> I'm guessing this applies _after_ the package is installed. Further, the message you give is a warning, not an error. Could you be clearer on what actually goes wrong when you don't have `$CUDA_PATH` set?

The very same error happens here. CUDA 11.5, 10.2 and pycuda 2021, 2020 and 2019 no combination works (python fixed to 3.6.9) under win10.
At first sight, everything installs fine (CUDA, pycuda), but importing pycuda.driver fails.

`import pycuda`  imports without errors

`import pycuda.driver` -> 
` File "<stdin>", line 1, in <module>
  File "C:\Users\xxxxxx\Anaconda3\envs\xxx\lib\site-packages\pycuda\driver.py", line 62, in <module>
    from pycuda._driver import *  # noqa
ImportError: DLL load failed: ~mdule not found~.`




`set CUDA` -> 
`CUDA_PATH=C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\NVIDIA Corporation\Nsight Compute 2021.3.1\;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.5\bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.5\libnvvp;c:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.2\bin;c:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.2\libnvvp;
CUDA_PATH_V10_2=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.2
CUDA_PATH_V11_5=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.5`


`C:\Users\xxxx\Anaconda3\envs\xxxxxx\lib\site-packages\pycuda\_driver.cp36-win_amd64.pyd`
exists. What is the problem. Can"t python load the pyd or does it not find cuda?

**cause of the error found: **  https://github.com/lucasgDependencies  showed, that `nvcuda.dll` could not be loaded. NVCUDA.dll is part of the graphics-card-driver provided by NVIDIA.  The graphics-card here is not supported by NVIDIA (despite CUDA installed without problems) - so there will be no NVCUDA.dll on this machine.
